### ISSUE TITLE
Single field validation now adds validation for all fields

### ISSUE THREAD

**Issue by olemartinorg** [Altinn/app-frontend-react#1472]
### Description of the bug

After #1281 it seems single field validation now shows all validation messages

### Steps To Reproduce

1. Open `krt/krt-1007a-1` as sophie salt + DDG fitness
2. Click `yes` in 1.2.3
3. Click `yes` in 1.4.1
4. This now un-hides `5-2-5-input-kolonne-a` which triggers single field validation and shows a ton of validation messages for unrelated fields (as these are not removed from the validation result before added to redux).

### Additional Information

- https://altinn.slack.com/archives/C02FK32KVKJ/p1695067297141039

**Comment by olemartinorg** [comment]
Hitting this a lot in #1442 now, and dynamics.ts (`Remove validation message when field disappears`) fails there often because of this (the error for the sum total in Grid is added even though we triggered single-field validation for the first name field. It most likely happens there more often since we made a lot of things faster - but I found that using `takeEvery`  made it fail a lot more often, since the `runSingleFieldValidationSaga` may get cancelled if there are more fields that trigger single field validation. Sooo, setting this should make the bug reproduce in our cypress tests as well:

```
      runSingleFieldValidation: mkAction<IRunSingleFieldValidation>({
        takeEvery: runSingleFieldValidationSaga,
      }),
```

@bjosttveit

**Comment by olemartinorg** [comment]
Closing as duplicate of #979
